File: Retos/Reto #40 - TABLA DE MULTIPLICAR [Fácil]/go/blackriper.go
package main

import "fmt"

// definir metodos de trabajos
type Table interface {
	ReadNumber()
	PrintTable()
}

// implementar metodos de trabajo
type Multiplication struct {
	Number int
}

func (m *Multiplication) ReadNumber() {
	fmt.Println("What multiplication table do you want to view?")
	fmt.Scanf("%d", &m.Number)
}

func (m *Multiplication) PrintTable() {
	fmt.Printf("Multiplication table  %v \n", m.Number)
	for n := 1; n <= 10; n++ {
		fmt.Printf("%v x %v = %v \n", m.Number, n, (m.Number * n))
	}
}

func main() {
	var multab Table = &Multiplication{}
	multab.ReadNumber()
	multab.PrintTable()
}

File: Retos/Reto #40 - TABLA DE MULTIPLICAR [Fácil]/typescript/blackriper.ts

// metodos de trabajo 
interface Table {
  read_number(num: number): void
  print_table(): void
}
// implementar metodos de trabajo
class MultiplicationTables implements Table {

  private nu: number

  read_number(num: number): void {
    this.nu = num
  }
  print_table(): void {
    for (let n = 1; n <= 10; n++) {
      console.log(`${this.nu} x ${n}= ${this.nu * n}`)
    }
  }

}

const multtab = new MultiplicationTables()
multtab.read_number(5)
multtab.print_table()


